### PR TITLE
update Web/API/URL/URL

### PR DESCRIPTION
**Pull request by jihuayu**

### CODE REVIEW

File: files/zh-cn/web/api/url/url/index.md [hunk 1]
```diff
@@ -5,24 +5,27 @@ slug: Web/API/URL/URL
...
 ### 参数
 
 - `url`
-  - : 是一个表示绝对或相对 URL 的 {{domxref("DOMString")}}。如果`url` 是相对 URL，则会将 `base` 用作基准 URL。如果 `url` 是绝对 URL，则无论参数`base`是否存在，都将被忽略。
+  - : 一个表示绝对或相对 URL 的 {{domxref("DOMString")}} 或任何具有字符串化方法的对象，如 <a>， <area> 元素。如果 `url` 是相对 URL，则会将 `base` 用作基准 URL。如果 `url` 是绝对 URL，则无论参数 `base` 是否存在，都将被忽略。
 - `base` {{optional_inline}}
-  - : 是一个表示基准 URL 的 {{domxref("DOMString")}}，在 _url_ 是相对 URL 时，它才会起效。如果未指定，则默认为 `''`。
+  - : 一个表示基准 URL 的 {{domxref("DOMString")}}，当 url 为相对 URL 时，它才会生效。如果未指定，它默认为 `undefined`。
+
+> **备注：** `url` 和 `base` 参数将从任意对象转换为字符串，就像其他接受字符串的 Web API 一样。特别地，您可以为任一参数使用现有的 {{domxref("URL")}} 对象，它将被转换为该对象的 {{domxref("URL.href", "href")}} 属性。
```
Comment (yin1999) [inline files/zh-cn/web/api/url/url/index.md:28]:
```suggestion
> **备注：** `url` 和 `base` 参数将从任意对象转换为字符串，就像其他接受字符串的 Web API 一样。特别地，你可以为任一参数使用现有的 {{domxref("URL")}} 对象，它将被转换为该对象的 {{domxref("URL.href", "href")}} 属性。
```

File: files/zh-cn/web/api/url/url/index.md [hunk 2]
```diff
@@ -33,24 +36,54 @@ const url = new URL(url [, base])
 ## 示例
 
 ```js
-let m = 'https://developer.mozilla.org';
-let a = new URL("/", m);                                // => 'https://developer.mozilla.org/'
-let b = new URL(m);                                     // => 'https://developer.mozilla.org/'
+// Base URLs:
```
Comment (yin1999) [inline files/zh-cn/web/api/url/url/index.md:39]:
```suggestion
// 基准 URL：
```

File: files/zh-cn/web/api/url/url/index.md [hunk 3]
```diff
@@ -33,24 +36,54 @@ const url = new URL(url [, base])
...
+new URL("/en-US/docs", D);
+// => 'https://developer.mozilla.org/en-US/docs'
+
+new URL("/en-US/docs", A);
+// => 'https://developer.mozilla.org/en-US/docs'
+
+new URL("/en-US/docs", "https://developer.mozilla.org/fr-FR/toto");
+// => 'https://developer.mozilla.org/en-US/docs'
+
+// Invalid URLs:
```
Comment (yin1999) [inline files/zh-cn/web/api/url/url/index.md:63]:
```suggestion
// 无效 URL：
```

File: files/zh-cn/web/api/url/url/index.md [hunk 4]
```diff
@@ -33,24 +36,54 @@ const url = new URL(url [, base])
...
+
+new URL("/en-US/docs", "");
+// Raises a TypeError exception as '' is not a valid URL
```
Comment (yin1999) [inline files/zh-cn/web/api/url/url/index.md:66]:
```suggestion
// 抛出 TypeError 异常：'' is not a valid URL
```

File: files/zh-cn/web/api/url/url/index.md [hunk 5]
```diff
@@ -33,24 +36,54 @@ const url = new URL(url [, base])
...
+
+new URL("/en-US/docs");
+// Raises a TypeError exception as '/en-US/docs' is not a valid URL
```
Comment (yin1999) [inline files/zh-cn/web/api/url/url/index.md:69]:
```suggestion
// 抛出 TypeError 异常：'/en-US/docs' is not a valid URL
```

File: files/zh-cn/web/api/url/url/index.md [hunk 6]
```diff
@@ -33,24 +36,54 @@ const url = new URL(url [, base])
...
+
+// Other cases:
```
Comment (yin1999) [inline files/zh-cn/web/api/url/url/index.md:71]:
```suggestion
// 其他示例：
```

File: files/zh-cn/web/api/url/url/index.md [hunk 7]
```diff
@@ -33,24 +36,54 @@ const url = new URL(url [, base])
...
-let d = new URL('/en-US/docs', b);                     // => 'https://developer.mozilla.org/en-US/docs'
-        new URL('/en-US/docs', d);                     // => 'https://developer.mozilla.org/en-US/docs'
-        new URL('/en-US/docs', a);                     // => 'https://developer.mozilla.org/en-US/docs'
+new URL("http://www.example.com", B);
+// => 'http://www.example.com/'
 
-        new URL('/en-US/docs', "https://developer.mozilla.org/fr-FR/toto");
-                                                       // => 'https://developer.mozilla.org/en-US/docs'
+new URL("", "https://example.com/?query=1");
+// => 'https://example.com/?query=1' (Edge before 79 removes query arguments)
```
Comment (yin1999) [inline files/zh-cn/web/api/url/url/index.md:80]:
```suggestion
// => 'https://example.com/?query=1'（Edge 早于 79 的版本会移除查询参数）
```

File: files/zh-cn/web/api/url/url/index.md [hunk 8]
```diff
@@ -33,24 +36,54 @@ const url = new URL(url [, base])
...
 
-        new URL('/en-US/docs', '');                    // Raises a TypeError exception as '' is not a valid URL
-        new URL('/en-US/docs');                        // Raises a TypeError exception as '/en-US/docs' is not a valid URL
-        new URL('http://www.example.com', );           // => 'http://www.example.com/'
-        new URL('http://www.example.com', b);          // => 'http://www.example.com/'
+new URL("/a", "https://example.com/?query=1");
+// => 'https://example.com/a' (see relative URLs)
```
Comment (yin1999) [inline files/zh-cn/web/api/url/url/index.md:83]:
```suggestion
// => 'https://example.com/a'（见相对 URL）
```

File: files/zh-cn/web/api/url/url/index.md [hunk 9]
```diff
@@ -33,24 +36,54 @@ const url = new URL(url [, base])
...
 
-        new URL("//foo.com", "https://example.com")    // => 'https://foo.com' (see relative URLs)
+new URL("//foo.com", "https://example.com");
+// => 'https://foo.com/' (see relative URLs)
```
Comment (yin1999) [inline files/zh-cn/web/api/url/url/index.md:86]:
```suggestion
// => 'https://foo.com/'（见相对 URL）
```

File: files/zh-cn/web/api/url/url/index.md [hunk 10]
```diff
@@ -63,4 +96,5 @@ let d = new URL('/en-US/docs', b);                     // => 'https://developer.
 
 ## 参见
 
+- [Polyfill of `URL` in `core-js`](https://github.com/zloirock/core-js#url-and-urlsearchparams)
```
Comment (yin1999) [inline files/zh-cn/web/api/url/url/index.md:99]:
```suggestion
- [`core-js` 中 `URL` 的 polyfill](https://github.com/zloirock/core-js#url-and-urlsearchparams)
```

**Review by yin1999 — COMMENTED** [review]
Some suggestions

File: files/zh-cn/web/api/url/url/index.md [hunk 11]
```diff
@@ -5,24 +5,27 @@ slug: Web/API/URL/URL
...
-如果给定的基本 URL 或生成的 URL 不是有效的 URL 链接，则会抛出一个{{jsxref("TypeError")}}。
+如果给定的基本 URL 或生成的 URL 不是有效的 URL，则会抛出 {{jsxref("TypeError")}} 异常。
 
 {{AvailableInWorkers}}
 
 ## 语法
 
-```
-const url = new URL(url [, base])
+```js
```
[the remark this answers is not in the document]
Comment (jihuayu) [reply files/zh-cn/web/api/url/url/index.md:16]: 请问下这个 `js-nolint` 和 `js` 有什么区别？

[the remark this answers is not in the document]
Comment (yin1999) [reply files/zh-cn/web/api/url/url/index.md]: See: https://developer.mozilla.org/zh-CN/docs/MDN/Writing_guidelines/Howto/Markdown_in_MDN#%E9%98%BB%E6%AD%A2_lint

We don't want the linter to format the code block (would be formated to):

```js
new URL(url);
new URL(url, base);
```

**Review by yin1999 — APPROVED** [review]
LGTM! Thank you :)

**Pull request merged**